sechub-commons-model/src/main/java/com/mercedesbenz/sechub/commons/model/SecHubConfigurationModelValidator.java

```diff
@@ -276,6 +278,49 @@ private void handleApi(InternalValidationContext context, SecHubWebScanConfigura
         handleUsages(context, openApi);
     }
 
+    private void handleHTTPHeaders(InternalValidationContext context, SecHubWebScanConfiguration webScan) {
+        Optional<List<HTTPHeaderConfiguration>> optHttpHeaders = webScan.getHeaders();
+        if (!optHttpHeaders.isPresent()) {
+            return;
+        }
+
+        List<HTTPHeaderConfiguration> httpHeaders = optHttpHeaders.get();
+        for (HTTPHeaderConfiguration httpHeader : httpHeaders) {
+            if (httpHeader.getName() == null || httpHeader.getName().isEmpty()) {
+                context.result.addError(WEB_SCAN_NO_HEADER_NAME_DEFINED);
+                return;
+            }
+            if (httpHeader.getValue() == null || httpHeader.getValue().isEmpty()) {
+                context.result.addError(WEB_SCAN_NO_HEADER_VALUE_DEFINED);
+                return;
+            }
+            if (!httpHeader.getOnlyForUrls().isPresent()) {
+                continue;
+            }
+            validateHTTPHeaderUrls(context, httpHeader.getOnlyForUrls().get(), webScan.getUrl().toString());
+        }
+
+    }
+
+    private void validateHTTPHeaderUrls(InternalValidationContext context, List<String> onlyForUrls, String targetUrl) {
+        for (String url : onlyForUrls) {
+            url = url.replaceAll(Pattern.quote(SecHubWebScanConfiguration.WEBSCAN_URL_WILDCARD_SYMBOL), "");
```

Comment:
```suggestion
        URL  onlyForUrlToCheck = createURLWithoutWildCardsAndTrailings(onlyForUrl)
```
I would seperate this